source/glkernel/include/glkernel/sample.hpp

```diff
@@ -207,6 +208,33 @@ size_t poisson_square(tkernel<glm::tvec2<T, P>> & kernel, const T min_dist, cons
     return k + 1;
 }
 
+template <typename T, glm::precision P>
+size_t n_rooks(tkernel<glm::tvec2<T, P>> & kernel)
+{
+	assert(kernel.depth() == 1);
+	assert(kernel.width() == kernel.height());
+
+	auto stratum_size = 1.0 / kernel.width();
+	std::random_device RD;
+	std::mt19937_64 generator(RD());
+	std::uniform_real_distribution<> jitter_dist(0.0, stratum_size);
+
+	std::vector<unsigned int> pool;
+	for (unsigned int k = 0; k < kernel.width(); ++k)
+	{
+		pool.push_back(k);
+	}
+	std::random_shuffle(pool.begin(), pool.end());
```

Comment:
hmm, :smile: perhaps we can use glkernel to do this in the future? we should make a decision on whether or not to use kernels internally for the manipulation of kernels ... in this case this is currently hindered due to missing support of integer coefficients in kernels ...